### PR TITLE
feat!: bump internal abi to 15

### PR DESCRIPTION
**Pull request by amaanq**

### CODE REVIEW

File: lib/include/tree_sitter/api.h [hunk 1]
```diff
@@ -26,13 +26,13 @@ extern "C" {
...
  */
-#define TREE_SITTER_LANGUAGE_VERSION 14
+#define TREE_SITTER_LANGUAGE_VERSION 15
 
 /**
  * The earliest ABI version that is supported by the current version of the
  * library.
  */
-#define TREE_SITTER_MIN_COMPATIBLE_LANGUAGE_VERSION 13
+#define TREE_SITTER_MIN_COMPATIBLE_LANGUAGE_VERSION 14
```
Comment (clason) [inline lib/include/tree_sitter/api.h:35]: Is this necessary? A number of parsers in the wild are still on ABI 13.

Comment (amaanq) [reply lib/include/tree_sitter/api.h:35]: ABI 14 is more than two years old at this point

Comment (clason) [reply lib/include/tree_sitter/api.h:35]: So? What harm does this do?

Comment (amaanq) [reply lib/include/tree_sitter/api.h:35]: Ok, we can keep the runtime support for 13, but I'll remove allowing the CLI to generate ABI 13. If the compatibility code between 14 and 15 becomes too unwieldy with 13 in the mix later on, I'll revisit removing runtime support then.

Comment (clason) [reply lib/include/tree_sitter/api.h]: Yes, that is a good compromise.

**Pull request merged**